pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -94,92 +89,103 @@ func (c *Cluster) configureDashboard(m *mgrConfig) error {
 }
 
 // Ceph docs about the dashboard module: http://docs.ceph.com/docs/nautilus/mgr/dashboard/
-func (c *Cluster) toggleDashboardModule(m *mgrConfig) error {
+func (c *Cluster) configureDashboardModules() error {
 	if c.dashboard.Enabled {
 		if err := client.MgrEnableModule(c.context, c.Namespace, dashboardModuleName, true); err != nil {
 			return fmt.Errorf("failed to enable mgr dashboard module. %+v", err)
 		}
-
-		if err := c.initializeSecureDashboard(); err != nil {
-			return fmt.Errorf("failed to initialize dashboard. %+v", err)
-		}
-
-		if err := c.configureDashboardModule(m); err != nil {
-			return fmt.Errorf("failed to configure mgr dashboard module. %+v", err)
-		}
 	} else {
 		if err := client.MgrDisableModule(c.context, c.Namespace, dashboardModuleName); err != nil {
 			logger.Errorf("failed to disable mgr dashboard module. %+v", err)
 		}
 	}
+
+	if !c.dashboard.Enabled {
+		return nil
```

Comment:
Shouldn't this be part of the `else` line 97 after running `client.MgrDisableModule()`?

Reply:
?